src/main/java/com/rapid7/client/dcerpc/io/PacketOutput.java

```diff
@@ -100,29 +110,48 @@ public void writeString(final String string, final boolean nullTerminate)
 
         if (string != null) {
             writeReferentID();
-            writeInt(maximumChars);
-            writeInt(0);
-            writeInt(currentChars);
-            writeChars(string);
-
-            if (nullTerminate) {
-                writeShort((short) 0);
-            }
-
-            align();
+            writeString(string, maximumChars, currentChars, nullTerminate);
         } else {
             writeNull();
         }
     }
 
     public void writeStringRef(final String string, final boolean nullTerminate)
         throws IOException {
-        if (string != null) {
-            writeReferentID();
-            writeString(string, nullTerminate);
+        writeReferentID();
+        writeString(string, nullTerminate);
+    }
+
+    public void writeString(final String string, final boolean nullTerminate)
+        throws IOException {
+        final int maximumChars;
+        final int currentChars;
+
+        if (string == null) {
```

Comment:
Why would we be marshaling for a null string?  It's null.  This write method shouldn't support a null string.

Reply:
This is because writeString(null) was a valid use case in legacy code.